lib/Service/AutoCompletion/AutoCompleteService.php

```diff
@@ -23,22 +24,28 @@
 
 use OCA\Mail\Db\CollectedAddress;
 use OCA\Mail\Service\ContactsIntegration;
+use OCA\Mail\Service\GroupsIntegration;
 
 class AutoCompleteService {
 
 	/** @var ContactsIntegration */
 	private $contactsIntegration;
 
+	/** @var GroupsIntegration */
+	private $groupsIntegration;
+
 	/** @var AddressCollector */
 	private $addressCollector;
 
-	public function __construct(ContactsIntegration $ci, AddressCollector $ac) {
+	public function __construct(ContactsIntegration $ci, GroupsIntegration $gi, AddressCollector $ac) {
 		$this->contactsIntegration = $ci;
+		$this->groupsIntegration = $gi;
 		$this->addressCollector = $ac;
 	}
 
 	public function findMatches($term) {
 		$recipientsFromContacts = $this->contactsIntegration->getMatchingRecipient($term);
+		$recipientGroups = $this->groupsIntegration->getMatchingGroups($term);
```

Comment:
I'd prefer to have this logic moved to a service class, maybe even introduce a facade for that. But that's a minor enhancement - this can be done later on :wink: